src/components/inputs/multi-select/multi-select.component.tsx

```diff
@@ -16,44 +16,39 @@ import styles from './multi-select.scss';
 const MultiSelect: React.FC<FormFieldProps> = ({ question, onChange, handler, previousValue }) => {
   const { t } = useTranslation();
   const [field, meta] = useField(question.id);
-  const { setFieldValue, encounterContext, layoutType, workspaceLayout } = React.useContext(FormContext);
+  const { setFieldValue, encounterContext, layoutType, workspaceLayout, isFieldInitializationComplete } =
+    React.useContext(FormContext);
   const [errors, setErrors] = useState([]);
   const [warnings, setWarnings] = useState([]);
   const [counter, setCounter] = useState(0);
-  const [touched, setTouched] = useState(false);
   const isFieldRequiredError = useMemo(() => errors[0]?.errCode == fieldRequiredErrCode, [errors]);
 
-  useEffect(() => {
-    // Carbon's MultiSelect has issues related to not updating the component based on the `initialSelectedItems` prop
-    // this is an intermittent issue. As a temporary solution, were are forcing the component to re-render
-    if (field.value && field.value.length == 0) {
-      // chances are high the value was cleared
-      // force the Multiselect component to be re-mounted
-      setCounter(counter + 1);
-    } else if (!touched && question.value) {
-      setCounter(counter + 1);
-    }
-  }, [field.value]);
-
   useEffect(() => {
     if (question['submission']) {
       question['submission'].errors && setErrors(question['submission'].errors);
       question['submission'].warnings && setWarnings(question['submission'].warnings);
     }
   }, [question['submission']]);
 
+  const selectOptions = question.questionOptions.answers
+    .filter((answer) => !answer.isHidden)
+    .map((answer, index) => ({
+      id: `${question.id}-${answer.concept}`,
+      concept: answer.concept,
+      label: answer.label,
+      key: index,
+      disabled: answer.disable?.isDisabled,
+    }));
+
   const initiallySelectedQuestionItems = useMemo(() => {
-    const selectedItems = [];
-    question.questionOptions.answers.forEach((item) => {
-      if (field.value?.includes(item.concept)) {
-        selectedItems.push(item);
-      }
-    });
-    return selectedItems;
-  }, [question, field.value]);
+    if (isFieldInitializationComplete && field.value?.length && counter < 1) {
+      setCounter(counter + 1);
+      return selectOptions.filter((item) => field.value?.includes(item.concept));
+    }
+    return [];
+  }, [isFieldInitializationComplete, field.value]);
```

Comment:
`counter`, `setCounter` and `selectOptions` will also be a dependency.

Reply:
Well yes but we don't need to subscribe to them to evaluate the `initiallySelectedQuestionItems`.
1. Subscribing to `selectOptions` introduces a weird bug (I had it as a dep in my initial implementation). 
2. We evaluate the state only if the `counter` is 0 so we don't have to subscribe to it.